src/Algebra/Module.agda

```diff
@@ -0,0 +1,285 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Definitions of algebraic structures defined over some other
+-- structure, like modules and vector spaces
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Algebra.Module where
+
+open import Algebra
+import Algebra.Structures.Module as Str
+open import Algebra.FunctionProperties.Core
+import Algebra.FunctionProperties.LeftAction as LFP
+import Algebra.FunctionProperties.RightAction as RFP
+open import Function
+open import Level
+open import Relation.Binary
+
+------------------------------------------------------------------------
+-- Left modules
+------------------------------------------------------------------------
+
+record LeftSemimodule {r ℓr} (semiring : Semiring r ℓr) m ℓm
+                      : Set (r ⊔ ℓr ⊔ suc (m ⊔ ℓm)) where
+  open Semiring semiring
+
+  infixr 7 _*ₗ_
+  infixl 6 _+ᴹ_
+  infix 4 _≈ᴹ_
+
+  field
+    Carrierᴹ : Set m
+    _≈ᴹ_ : Rel Carrierᴹ ℓm
+  open Str _≈ᴹ_
+  field
+    _+ᴹ_ : Op₂ Carrierᴹ
+    _*ₗ_ : Opₗ Carrier Carrierᴹ
+    0ᴹ : Carrierᴹ
+    isLeftSemimodule : IsLeftSemimodule semiring _+ᴹ_ _*ₗ_ 0ᴹ
+
+  open IsLeftSemimodule isLeftSemimodule public
+
+  +ᴹ-commutativeMonoid : CommutativeMonoid m ℓm
+  +ᴹ-commutativeMonoid =
+    record { isCommutativeMonoid = +ᴹ-isCommutativeMonoid }
```

Comment:
The record declarations should conform to the [style guide](https://github.com/agda/agda-stdlib/blob/master/notes/style-guide.md).